cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -1103,6 +1399,115 @@ const FunctionDoc assume_timezone_doc{
     {"timestamps"},
     "AssumeTimezoneOptions"};
 
+const FunctionDoc years_between_doc{
+    "Compute the number of years between two timestamps",
+    ("Returns the number of year boundaries crossed from the first timestamp to the\n"
```

Comment:
Should `\n`s appear mid sentence? Same below.

Reply:
I think it's fine, e.g. see https://github.com/apache/arrow/blob/5260fd5970d34c62e0ae7680f606b29eab56d96b/cpp/src/arrow/compute/kernels/vector_nested.cc#L141 or https://github.com/apache/arrow/blob/5260fd5970d34c62e0ae7680f606b29eab56d96b/cpp/src/arrow/compute/kernels/vector_sort.cc#L1643-L1650